NAAM.md

```diff
@@ -0,0 +1,278 @@
+# Naam: Naming As Advertisement
+
+![wip](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
+
+**Author(s)**:
+
+- [Masih Derkani](https://github.com/masih)
+
+**Maintainer(s)**:
+
+- [Masih Derkani](https://github.com/masih)
+
+* * *
+
+**Abstract**
+
+This document defines the **N**aming **A**s **A**dvertise**m**ent (Naam) protocol: a mechanism by
+which [IPNI](IPNI.md) can be used to both publish, and
+resolve [IPNS](https://github.com/ipfs/specs/blob/main/ipns/IPNS.md) records. Naam utilises the
```

Comment:
IIUC what you've identified here is reasonably that you can use this mechanism to store basically any type of small data. In particular, you don't actually need to change how any IPNI node operates in order for this proposal to function.

Instead, the proposal is more about letting people know what to do consistently so that they can coordinate across applications in the same way that the Bitswap and FilecoinGraphSync-v1 codes and metadata are defined so as to increase coordination. Also, it allows IPNI nodes to use APIs like https://github.com/ipfs/specs/pull/337 reasonably instead of only responding to `FindProviders` queries and having clients decipher them on the their end.

I don't know if down the road advertising chains and networks might want to be divided up more so people can choose to participate in them more selectively, but without damaging the system as a whole. Doesn't seem necessary now or for this PR (given that it already works as is), but perhaps something to consider as the system and the data it stores grows.

Reply:
Spot on; that's exactly why this spec exists.

On separation of networks; maybe down the line. It seems like a big overkill to just separate for IPNS records considering how few of them are out there compared to content CIDs.